### ISSUE TITLE
MARC import, get Author from 700 if no 1xx exists

### ISSUE THREAD

**Issue by hornc** [internetarchive/openlibrary#1530]
https://github.com/internetarchive/openlibrary/blob/master/openlibrary/catalog/marc/parse.py#L323

`read_authors()` only reads 1XX fields to determine author

There are... in progress?... comments implying 7xx fields should be used if there is no 1xx

```
# 1. if authors in 100, 110, 111 use them
# 2. if first contrib is 710 or 711 use it
# 3. if
```
Here is an example MARC record with no 1XX author: https://ia601304.us.archive.org/32/items/sexuallytransmit00egen/sexuallytransmit00egen_marc.xml

```
<datafield tag="700" ind1="1" ind2=" ">
<subfield code="a">Egendorf, Laura K.,</subfield>
<subfield code="d">1973-</subfield>
</datafield>
```
Indicating the editor (rather than author). This was imported as https://openlibrary.org/works/OL17937580W/Sexually_transmitted_diseases  (Unknown Author)

The editor exists on OL as https://openlibrary.org/authors/OL2733059A/Laura_K._Egendorf

**TASK:** clean up the comments and begin to make use of 7XX values if no 1XX are present.

**Comment by tfmorris** [comment]
I think this would be a useful improvement to do before starting bulk imports again, but this particular record is always going to be problematic. The 700 has no $e relator field to indicate that she's the editor, so we'd have to do some type of smart inference using the freeform 245$c which seems like it could potential cause more problems than it fixes.

```
<datafield tag="245" ind1="0" ind2="0">
  <subfield code="a">Sexually transmitted diseases /</subfield>
  <subfield code="c">Laura Egendorf, book editor.</subfield>
</datafield>
```

Using 700s which have relators of editor, compiler, etc seems like it could be a useful incremental addition though.

**Comment by hornc** [comment]
This looks like it may be fixed since the original import of that book was performed

I imported this book to my local dev environment: 
` r = ol.session.post(ol.base_url + '/api/import/ia', data={'identifier': 'sexuallytransmit00egen'})`

and it picked up the following data from the original MARC, the author's name is only in the 700 field, so it is picking it up, at least for a new import?

```
{'comment': 'import new book', 'data': '{}', 'query': '[{"type": {"key": "/type/author"}, "name": "Egendorf, Laura K., 1973-", "key": "/authors/OL10A"}, {"title": "Sexually transmitted diseases", "covers": [null], "subjects": ["Sexually transmitted diseases"], "key": "/works/OL2W", "authors": [{"type": {"key": "/type/author_role"}, "author": {"key": "/authors/OL10A"}}], "type": {"key": "/type/work"}}, {"source_records": ["ia:sexuallytransmit00egen"], "language": "eng", "title": "Sexually transmitted diseases", "openlibrary": "OL23139321M", "covers": [null], "subjects": ["Sexually transmitted diseases"], "key": "/books/OL10M", "authors": [{"key": "/authors/OL10A"}], "ocaid": "sexuallytransmit00egen", "works": [{"key": "/works/OL2W"}], "type": {"key": "/type/edition"}}]'}
```

**Comment by tfmorris** [comment]
This author record from 2008: https://openlibrary.org/authors/OL363064A.json?v=1
got imported from this MARC: https://openlibrary.org/show-records/marc_university_of_toronto/uoft.marc:338337433:849

including the birth year, but not including the 700$q "Fuller form of name" subfield, which is an unfortunate gap. We should include the $q subfield for not only 700s, but 600, 100, and any other relevant personal name fields, probably putting them in the AKA field.

**Comment by tfmorris** [comment]
@hornc Do you still think the problem as originally stated exists? I'm guessing not, but there seem to be some loose ends which could be cleaned up (in addition to the comments that you already improved).

Should we repurpose this ticket for the missing pieces or close it and start fresh? I don't feel strongly one way or the other.

In addition to the 700$q fuller name handling in my example, it looks like your original example is missing the structured birth date and has the author name in inverted form, so those need to be cleaned up too.

**Comment by hornc** [comment]
@tfmorris I have just retested this and added a unit test to cover this case in the PR #2101 . It looks solved, and the test should prove and protect the functionality. 

None of the existing code looks for the $q subfield for any author, so I have raised it as a separate issue to be done for 100, and 700s , and pointed to where the relevant code is rather than adding the feature in with the tests PR. Other 1xx (e.g. organisations) don't seem to have the Fuller Name sub field, and we're not making use of 600 / people in this way, so I _think_ 100 + 700 will be sufficient, and can be added in the same method. Good idea anyway, and feel free to comment on the new #2103 if I have missed anything!
